core/identity/src/service/identity.rs

```diff
@@ -504,3 +526,86 @@ impl IdentityService {
         });
     }
 }
+
+pub async fn wait_for_default_account_unlock(db: &DbExecutor) -> anyhow::Result<()> {
+    let identity_key = get_default_identity_key(&db).await?;
+
+    if identity_key.is_locked() {
+        log::info!("{}", yansi::Color::RGB(0xFF, 0xA5, 0x00).paint(
+            "Daemon cannot start because default account is locked. Unlock it by running 'yagna id unlock'"
+        ));
+
+        let locked_identity = identity_key.id();
+        let (tx, rx) = futures::channel::mpsc::unbounded();
+        let endpoint = format!("{}/await_unlock", model::BUS_ID);
+
+        let _ = bus::bind(&endpoint, move |e: model::event::Event| {
+            let mut tx_clone = tx.clone();
+            async move {
+                match e {
+                    model::event::Event::AccountLocked { .. } => {}
+                    model::event::Event::AccountUnlocked { identity } => {
+                        if locked_identity == identity {
+                            log::debug!("Got unlocked event for default locked account");
+                            tx_clone.send(()).await.expect("Receiver is closed");
+                        }
+                    }
+                };
+                Ok(())
+            }
+        });
+        subscribe(endpoint.clone()).await?;
+
+        wait_for_unlock(rx).await?;
```

Comment:
We should check `get_default_identity_key` for the second time here.
User could unlock his account, after the first check but before you subscribed to events and were able to capture event